### PR TITLE
Make client authentication clearer

### PR DESCRIPTION
**Pull request by nbarbettini**
Fixed some problems as I was reviewing the OIDC API reference.

- [x] These endpoints (token, introspect, etc.) require _authentication_, not specifically _token authentication_. That was incorrect.
- [x] Clarified that you use client authentication when you _call_ the endpoint, but which authentication method you choose is determined when you create the application in Okta (one-time setup). This wasn't clear to me before.
- [x] Rearranged some text to make it easier to read.

@susanharper-okta Can you give this a writer sanity check, and coordinate with the Fed team to review? Thank you!

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/api/resources/oidc/index.md [hunk 1]
```diff
@@ -1203,9 +1203,12 @@ For more information about configuring an app for OpenID Connect, including grou
...
 
 ## Client Authentication Methods
-When registering an OAuth 2 client application, you can specify an authentication method by including the [token_endpoint_auth_method](https://developer.okta.com/docs/api/resources/apps/#add-oauth-2-0-client-application) parameter. To decide which method to use, it depends on what type of client app you are registering and what that client app is able to provide. For example, you might be using a client that was built by a third party, and it might only support Basic Authentication. So, you would need to use the `client_secret_basic` client authentication method.
 
-> Native applications shouldn't provide -- and by default don't store -- the `client_secret` (see [Section 5.3.1 of the OAuth 2.0 spec](https://tools.ietf.org/html/rfc6819#section-5.3.1)). They can omit the `client_secret` from the request parameters when introspecting a token.
+When making requests to endpoints on this page that require authentication, you must authenticate the client (your application code) with Okta. This is done by including a header or parameter in the request, depending on the authentication method the application is configured with.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/api/resources/oidc/index.md]: This isn't clear to me: (your application code) - what are you trying to say here? What is an application code?

**Comment by nbarbettini** [comment]
Thanks @susanharper-okta and @danielzeller-okta. I made some more changes that incorporate your feedback. Can you take another look?

File: packages/@okta/vuepress-site/docs/api/resources/oidc/index.md [hunk 2]
```diff
@@ -1197,24 +1197,23 @@ For more information about configuring an app for OpenID Connect, including grou
...
 
 ## Client Authentication Methods
-When registering an OAuth 2 client application, you can specify an authentication method by including the [token_endpoint_auth_method](https://developer.okta.com/docs/api/resources/apps/#add-oauth-2-0-client-application) parameter. To decide which method to use, it depends on what type of client app you are registering and what that client app is able to provide. For example, you might be using a client that was built by a third party, and it might only support Basic Authentication. So, you would need to use the `client_secret_basic` client authentication method.
 
-> Native applications shouldn't provide -- and by default don't store -- the `client_secret` (see [Section 5.3.1 of the OAuth 2.0 spec](https://tools.ietf.org/html/rfc6819#section-5.3.1)). They can omit the `client_secret` from the request parameters when introspecting a token.
+Some endpoints require client authentication. To make requests to these endpoints, you must include a header or parameter in the request depending on the authentication method the application is configured with.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/api/resources/oidc/index.md]: For translation, add "that"

Some endpoints require client authentication. To make requests to these endpoints, you must include a header or parameter in the request depending on the authentication method that the application is configured with.

File: packages/@okta/vuepress-site/docs/api/resources/oidc/index.md [hunk 3]
```diff
@@ -1197,24 +1197,23 @@ For more information about configuring an app for OpenID Connect, including grou
...
 
-* `private_key_jwt`: Use this method when you want to use public/private key pairs for more security. The main benefit of this method is you can generate the private key on your own servers and never have it leave there for any reason, since you only need to provide the public key to Okta. This is better than `client_secret_jwt` since Okta must know what the `client_secret` string is beforehand, so there are more places that it could in theory be compromised. 
+* `client_secret_basic`, `client_secret_post`, `client_secret_jwt`: Use one of these methods when the client has a client secret. Public clients (such as single-page and mobile apps) that cannot protect a client secret must use `none` below.
 
-* `none` - Use this method when the client doesn't authenticate itself to the `/token` endpoint because it uses the [Implicit Flow](/authentication-guide/implementing-authentication/implicit/) or because it is a public client with no client secret or other authentication mechanism.
+* `private_key_jwt`: Use this when you want maximum security. This method is more complex and requires a server, so it also cannot be used with public clients.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/api/resources/oidc/index.md]: Try to make it more conversational in tone by using common contractions:

 `private_key_jwt`: Use this when you want maximum security. This method is more complex and requires a server, so it also can't be used with public clients.

File: packages/@okta/vuepress-site/docs/api/resources/oidc/index.md [hunk 4]
```diff
@@ -1197,24 +1197,23 @@ For more information about configuring an app for OpenID Connect, including grou
...
 
-> Note: If you don't specify a method when registering your client, the default method is `client_secret_basic`.
+* `none` - Use this with clients that do not have a client secret (such as applications that use the [authorization code flow with PKCE](/authentication-guide/implementing-authentication/auth-code-pkce/) or the [implicit flow](/authentication-guide/implementing-authentication/implicit/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/api/resources/oidc/index.md]: Try to make it more conversational in tone by using common contractions and close paren:

* `none` - Use this with clients that don't have a client secret (such as applications that use the [authorization code flow with PKCE](/authentication-guide/implementing-authentication/auth-code-pkce/) or the [implicit flow](/authentication-guide/implementing-authentication/implicit/)).

File: packages/@okta/vuepress-site/docs/api/resources/oidc/index.md [hunk 5]
```diff
@@ -1197,24 +1197,23 @@ For more information about configuring an app for OpenID Connect, including grou
...
+> Note: If you don't specify a method when registering your client, the default method is `client_secret_basic`.
 
-* `client_secret_basic`, `client_secret_post`, `client_secret_jwt`: Use one of these methods to authenticate against any endpoint that requires client authentication, when the client has access to the `client_secret`.
+Okta supports the following authentication methods, detailed in the sections below:
 
-* `private_key_jwt`: Use this method when you want to use public/private key pairs for more security. The main benefit of this method is you can generate the private key on your own servers and never have it leave there for any reason, since you only need to provide the public key to Okta. This is better than `client_secret_jwt` since Okta must know what the `client_secret` string is beforehand, so there are more places that it could in theory be compromised. 
+* `client_secret_basic`, `client_secret_post`, `client_secret_jwt`: Use one of these methods when the client has a client secret. Public clients (such as single-page and mobile apps) that cannot protect a client secret must use `none` below.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/api/resources/oidc/index.md]: Try to make it more conversational in tone by using common contractions:

* `client_secret_basic`, `client_secret_post`, `client_secret_jwt`: Use one of these methods when the client has a client secret. Public clients (such as single-page and mobile apps) that can't protect a client secret must use `none` below.

**Comment by nbarbettini** [comment]
Thanks all - @susanharper-okta if this looks good now, can you merge?

**Pull request merged**